src/isoldated-hybrid.store.spec.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  IsolatedHybridConfigurationStore,
+  ServingStoreUpdateStrategy,
+} from './isolated-hybrid.store';
+
+describe('IsolatedHybridConfigurationStore', () => {
+  const syncStoreMock = {
+    get: jest.fn(),
+    getKeys: jest.fn(),
+    isInitialized: jest.fn(),
+    setEntries: jest.fn(),
+  };
+
+  const asyncStoreMock = {
+    getEntries: jest.fn(),
+    isInitialized: jest.fn(),
+    isExpired: jest.fn(),
+    setEntries: jest.fn(),
+  };
+
+  const servingStoreUpdateStrategies: ServingStoreUpdateStrategy[] = ['always', 'expired', 'empty'];
+
+  // Dynamically create a describe() block for stores initialized with each of the three update strategies
+  servingStoreUpdateStrategies.forEach((updateStrategy) => {
```

Comment:
Non-blocking/nit: I've found wrapping tests in `forEach` blocks can make it difficult to focus on a test if something is failing. While not an issue here, it's also easy to forget to make sure the test names are differentiated via string interpolation. I generally prefer a little repetition with helper functions to centralize code. (none of that is really that big of a deal though, so there's really no need to spend time to make changes)

Reply:
Oh yeah, jest also has `describe.each` and `it.each` built in to it.

Reply:
I know about `it.each()` but didn't know about `describe.each()` -- I'll use that!

Agree on your concern that it can be hard to know which test is failing but I think putting the interpolation in is worth the simplicity of using the data provider.